src/libraries/System.IO.Pipelines/tests/PipeReaderReadAtLeastAsyncTests.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO.Pipelines.Tests.Infrastructure;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class ReadAtLeastAsyncTests
```

Comment:
There's a test you should move:
- https://github.com/dotnet/runtime/blob/98ac23212e6017c615e7e855e676fc43c8e44cb8/src/libraries/System.IO.Pipelines/tests/ReadAsyncCancellationTests.cs#L374
- https://github.com/dotnet/runtime/blob/98ac23212e6017c615e7e855e676fc43c8e44cb8/src/libraries/System.IO.Pipelines/tests/ReadAsyncCancellationTests.cs#L445
-

Reply:
I have moved the first one.
The second one is failing with the `StreamPipeReader` implementation as the returned (cancelled) buffer is empty.

Reply:
Sure. But I do not know how to fix it 🤔
Even if I bypass the `TryRead` canceled result, `InnerStream.ReadAsync` throws, as expected with a canceled token, an `OperationCanceledException` and the buffer remains empty...
Do I miss something here?

Reply:
I think you're right. Adjust the test to no expect the data to be there. Cancellation with the StreamPipeReader is dubious generally.

Reply:
Done.